master/buildbot/process/build.py

```diff
@@ -560,7 +626,9 @@ def buildFinished(self, text, results):
             # XXX: also test a 'timing consistent' flag?
             log.msg(" setting expectations for next time")
             self.builder.setExpectations(self.progress)
-        reactor.callLater(0, self.releaseLocks)
+        # TODO: check why callbacks are not triggered when stopping master and retrying builds
+        # this maybe cause by the upstream merged
+        eventually(self.releaseLocks)
```

Comment:
that was introduced by upstream, to handle the call queue and behind the scenes it calls the reactor.callLater from twisted, we adjusted the stats and tested with this changed and they are working fine.